src/luarocks/build/builtin.lua

```diff
@@ -143,12 +143,20 @@ local function compile_library(variables, flags)
          return ok, wrapname
       end
    else
-      compile.object = function(object, source, defines, incdirs)
+      compile.dynamic_object = function(object, source, defines, incdirs)
          local extras = {}
          add_flags(extras, "-D%s", defines, variables)
          add_flags(extras, "-I%s", incdirs, variables)
          return execute(variables.CC.." "..variables.CFLAGS, "-I"..variables.LUA_INCDIR, "-c", source, "-o", object, unpack(extras))
       end
+      if flags["static"] then
```

Comment:
You can create this function unconditionally. Only the calling of it needs to be conditional on `flags`.